# Patch Makefile.common.in to include -DEXPORT_SYMTAB and the various modules 
# to define their license type - to avoid a "kernel tainted" warning.

diff -U2 -r cryptoapi-2.4.7.0.old/Makefile.common.in cryptoapi-2.4.7.0/Makefile.common.in
--- cryptoapi-2.4.7.0.old/Makefile.common.in	Sun Jul 22 00:36:58 2001
+++ cryptoapi-2.4.7.0/Makefile.common.in	Wed Mar 20 13:44:00 2002
@@ -3,5 +3,5 @@
 # dynamic modules
 $(KMODULES): %.o: %.c
-	$(KCC) -I$(top_srcdir)/include -D__KERNEL__ -DMODULE $(KCPPFLAGS) $(KCFLAGS) -o $@ -c $<
+	$(KCC) -I$(top_srcdir)/include -D__KERNEL__ -DEXPORT_SYMTAB -DMODULE $(KCPPFLAGS) $(KCFLAGS) -o $@ -c $<
 
 # static modules
diff -U2 -r cryptoapi-2.4.7.0.old/api/cryptoapi.c cryptoapi-2.4.7.0/api/cryptoapi.c
--- cryptoapi-2.4.7.0.old/api/cryptoapi.c	Mon Jul  9 17:27:57 2001
+++ cryptoapi-2.4.7.0/api/cryptoapi.c	Wed Mar 20 13:44:00 2002
@@ -18,4 +18,10 @@
 #endif 
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 #include <linux/init.h>
 #include <linux/config.h>
diff -U2 -r cryptoapi-2.4.7.0.old/api/cryptoloop.c cryptoapi-2.4.7.0/api/cryptoloop.c
--- cryptoapi-2.4.7.0.old/api/cryptoloop.c	Sat Jul 21 23:52:46 2001
+++ cryptoapi-2.4.7.0/api/cryptoloop.c	Wed Mar 20 13:44:00 2002
@@ -37,4 +37,10 @@
 #endif 
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 #include <linux/init.h>
 #include <linux/config.h>
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-aes.c cryptoapi-2.4.7.0/ciphers/cipher-aes.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-aes.c	Sat Jul 21 23:33:26 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-aes.c	Wed Mar 20 13:44:00 2002
@@ -46,4 +46,10 @@
 #define D_KEY ((u32 *) (cx->keyinfo + 60))
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 static u8  pow_tab[256];
 static u8  log_tab[256];
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-blowfish.c cryptoapi-2.4.7.0/ciphers/cipher-blowfish.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-blowfish.c	Sun Jul 22 13:18:16 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-blowfish.c	Wed Mar 20 13:44:00 2002
@@ -7,4 +7,10 @@
 #define Bswap(x) __le32_to_cpu(x)
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 static u32 bf_pbox[16 + 2] = {
 	0x243f6a88, 0x85a308d3, 0x13198a2e, 0x03707344,
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-des-ede3.c cryptoapi-2.4.7.0/ciphers/cipher-des-ede3.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-des-ede3.c	Mon Jul  9 17:27:58 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-des-ede3.c	Wed Mar 20 13:44:00 2002
@@ -53,4 +53,10 @@
 #define ROR(d,c,o)      d = d >> c | d << o
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 typedef u8 DesData[8];
 typedef u32 DesKeys[32];
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-des.c cryptoapi-2.4.7.0/ciphers/cipher-des.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-des.c	Mon Jul  9 17:27:58 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-des.c	Wed Mar 20 13:44:00 2002
@@ -20,4 +20,10 @@
 #define ROR(d,c,o)      d = d >> c | d << o
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 typedef u8 DesData[8];
 typedef u32 DesKeys[32];
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-dfc.c cryptoapi-2.4.7.0/ciphers/cipher-dfc.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-dfc.c	Sat Jul 21 23:28:25 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-dfc.c	Wed Mar 20 13:44:00 2002
@@ -123,4 +123,10 @@
 #define BYTE_SWAP
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 /* The following arrays are all stored in big endian    */
 /* format with 32 bit words at lower array positions    */
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-dummy.c cryptoapi-2.4.7.0/ciphers/cipher-dummy.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-dummy.c	Mon Jul  9 17:27:58 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-dummy.c	Wed Mar 20 13:44:00 2002
@@ -15,4 +15,10 @@
 #include <linux/string.h>
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 static int
 dummy_encrypt (struct cipher_context *cx, const u8 *in8, u8 *out8,
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-idea.c cryptoapi-2.4.7.0/ciphers/cipher-idea.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-idea.c	Mon Jul  9 17:27:58 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-idea.c	Wed Mar 20 13:44:00 2002
@@ -69,4 +69,11 @@
 
 #define low16(x) ((x) & 0xFFFF)
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 typedef unsigned char byte;
 
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-mars.c cryptoapi-2.4.7.0/ciphers/cipher-mars.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-mars.c	Sun Aug  5 10:06:28 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-mars.c	Wed Mar 20 13:44:00 2002
@@ -42,4 +42,11 @@
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
+
 const static u32 s_box[] = {
 	0x09d0c479, 0x28c8ffe0, 0x84aa6c39, 0x9dad7287,	/* 0x000 */
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-rc5.c cryptoapi-2.4.7.0/ciphers/cipher-rc5.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-rc5.c	Sat Jul 21 23:36:43 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-rc5.c	Wed Mar 20 13:44:00 2002
@@ -75,4 +75,10 @@
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 /* Sets RC5 key */
 
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-rc6.c cryptoapi-2.4.7.0/ciphers/cipher-rc6.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-rc6.c	Sat Jul 21 23:36:46 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-rc6.c	Wed Mar 20 13:44:00 2002
@@ -64,4 +64,11 @@
 		a = rotr(a - l_key[i], u) ^ t
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
+
 /* initialise the key schedule from the user supplied key	*/
 
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-serpent.c cryptoapi-2.4.7.0/ciphers/cipher-serpent.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-serpent.c	Sun Jul 22 00:30:17 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-serpent.c	Wed Mar 20 13:44:00 2002
@@ -23,4 +23,10 @@
 #include <asm/byteorder.h>
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 #define rotl(reg, val) ((reg << val) | (reg >> (32 - val)))
 #define rotr(reg, val) ((reg >> val) | (reg << (32 - val)))
diff -U2 -r cryptoapi-2.4.7.0.old/ciphers/cipher-twofish.c cryptoapi-2.4.7.0/ciphers/cipher-twofish.c
--- cryptoapi-2.4.7.0.old/ciphers/cipher-twofish.c	Mon Aug 13 11:57:21 2001
+++ cryptoapi-2.4.7.0/ciphers/cipher-twofish.c	Wed Mar 20 13:44:00 2002
@@ -44,4 +44,10 @@
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,9)
+#ifdef MODULE_LICENSE
+MODULE_LICENSE("GPL");
+#endif
+#endif
+
 /* Structure for an expanded Twofish key.  s contains the key-dependent
  * S-boxes composed with the MDS matrix; w contains the eight "whitening"
Only in cryptoapi-2.4.7.0.old/: cryptoapi-2.4.7.0.patch
